CoursesManager.UI/Models/Repositories/CourseRepository/CourseRepository.cs

```diff
@@ -1,63 +1,71 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using System.Linq;
+using CoursesManager.MVVM.Data;
+using CoursesManager.MVVM.Env;
+using MySql.Data.MySqlClient;
 using CoursesManager.UI.Models;
 
 namespace CoursesManager.UI.Models.Repositories.CourseRepository
 {
-    public class CourseRepository : ICourseRepository
+    public class CourseRepository : BaseRepository, ICourseRepository
     {
-        private readonly ObservableCollection<Course> _courses;
-
-        public CourseRepository(ObservableCollection<Course> courses)
-        {
-            _courses = courses ?? throw new ArgumentNullException(nameof(courses), "Courses collection cannot be null.");
-        }
+        public CourseRepository() : base("courses", EnvManager<EnvModel>.Values.ConnectionString) {}
```

Comment:
Wellicht mooier om ipv in elke repo de connection string te specificeren dit in de baserepository te doen?

Dan misschien ook nog de connection string uitsplitten tot de benodige variablen en dan deze samen te bundelen in de base repository?

Dus ipv van connectionstring=

krijg je het volgende:

Host=
Port=
Username=
Password=

Reply:
Mee eens